File: doc-examples/example-python/src/test/python/example/BlobEventListener.py
from jakarta.inject import Singleton
from micronaut.context.event import ApplicationEventListener

from example.BlobFunction import BlobEvent


@Singleton
class BlobEventListener(ApplicationEventListener[BlobEvent]):
    received: list[str] = []

    def onApplicationEvent(self, event: BlobEvent) -> None:
        self.received = self.received + [event.content]

File: doc-examples/example-java/src/test/java/example/support/BlobEventListener.java
package example.support;

import example.BlobFunction;
import io.micronaut.context.event.ApplicationEventListener;
import jakarta.inject.Singleton;

import java.util.List;
import java.util.concurrent.CopyOnWriteArrayList;

@Singleton
public class BlobEventListener implements ApplicationEventListener<BlobFunction.BlobEvent> {

    private final List<String> received = new CopyOnWriteArrayList<>();

    @Override
    public void onApplicationEvent(BlobFunction.BlobEvent event) {
        received.add((String) event.getSource());
    }

    public List<String> getReceived() {
        return received;
    }
}
